src/rastervision/common/data/generators.py

```diff
@@ -66,11 +67,22 @@ def __init__(self, active_input_inds, train_ratio, cross_validation):
         if self.cross_validation is not None:
             self.process_cross_validation()
 
-        gen = self.make_split_generator(
-            TRAIN, target_size=(10, 10), batch_size=100, shuffle=True,
-            augment=False, normalize=False, only_xy=False)
-        batch = next(gen)
-        self.normalize_params = get_channel_stats(batch.all_x)
+        """
```

Comment:
Consider putting this stuff in a method called `get_channel_stats` which is distinguished from your other method which actually writes stats to disk.